### PR TITLE
Unit test cases for combining platform network reconciler into host reconciler for Day 1 operations

### PR DESCRIPTION
**Pull request by MahalakshmiManthiram**
Test plan:
1. 'Make test' runs successfully
2. Verify reconfiguration is blocked for all networks
3. Verify fresh configuration is successful for networks other than oam/admin/mgmt
4. Verify fresh configuration is bloced for oam/admin/mgmt networks
5. Verify ipv6 addresspool spec with pxeboot network fails to reconcile
6. Verify oam fails to reconcile if 'floating address' or 'gateway' in addresspool spec is missing
7. Verify pxeboot fails to reconcile if 'floating address' or 'controller0 address' is missing in addresspool spec

### CODE REVIEW

File: controllers/host/host_suite_test.go [hunk 1]
```diff
@@ -1,5 +1,5 @@
-/* SPDX-License-Identifier: Apache-2.0 */
-/* Copyright(c) 2019-2022 Wind River Systems, Inc. */
+// /* SPDX-License-Identifier: Apache-2.0 */
+// /* Copyright(c) 2019-2022 Wind River Systems, Inc. */
```
Comment (sriram-gn) [inline controllers/host/host_suite_test.go:2]: `//` can be removed and `/*` be retained.
Also please update the copyright notice -> `2019-2024`

File: controllers/platformnetwork_controller_test.go [hunk 2]
```diff
@@ -7,173 +7,171 @@ import (
...
-	"k8s.io/apimachinery/pkg/types"
-
-	"strings"
-
-	starlingxv1 "github.com/wind-river/cloud-platform-deployment-manager/api/v1"
+	// . "github.com/onsi/gomega"
+	// metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	// "k8s.io/apimachinery/pkg/types"
+	// "strings"
+	// starlingxv1 "github.com/wind-river/cloud-platform-deployment-manager/api/v1"
```
Comment (sriram-gn) [inline controllers/platformnetwork_controller_test.go:14]: Please remove if not used.

File: controllers/platformnetwork_controller_test.go [hunk 3]
```diff
@@ -7,173 +7,171 @@ import (
...
 	// comm "github.com/wind-river/cloud-platform-deployment-manager/common"
-	cloudManager "github.com/wind-river/cloud-platform-deployment-manager/controllers/manager"
+	// cloudManager "github.com/wind-river/cloud-platform-deployment-manager/controllers/manager"
```
Comment (sriram-gn) [inline controllers/platformnetwork_controller_test.go:16]: Please remove if not used.

File: controllers/platformnetwork_controller_test.go [hunk 4]
```diff
@@ -7,173 +7,171 @@ import (
...
+// 		return (host_instance.Status.StrategyRequired == expect_strategy)
+// 	}, timeout, interval).Should(BeTrue())
+
+// 	if host_instance.Status.StrategyRequired == cloudManager.StrategyLockRequired {
+// 		HostsListBodyResponse = strings.Replace(HostsListBody, `"administrative": "unlocked",`, `"administrative": "locked",`, 1)
+// 		HostsListBodyResponse = strings.Replace(HostsListBodyResponse, `"operational": "enabled"`, `"operational": "disabled"`, 1)
+// 	} else {
+// 		HostsListBodyResponse = HostsListBody
+// 	}
+// }
```
Comment (sriram-gn) [inline controllers/platformnetwork_controller_test.go:174]: Please remove commented part.
Request to create a single test case where we just create the `PlatformNetwork` resource.
Expect `Reconciled = InSync = false`

File: controllers/suite_test.go [hunk 5]
```diff
@@ -96,29 +95,29 @@ var _ = BeforeSuite(func() {
...
+	// err = (&PlatformNetworkReconciler{
+	// 	Client: k8sManager.GetClient(),
+	// 	Scheme: k8sManager.GetScheme(),
+	// }).SetupWithManager(k8sManager)
+	// // AddressPool
+	// err = (&AddressPoolReconciler{
+	// 	Client: k8sManager.GetClient(),
+	// 	Scheme: k8sManager.GetScheme(),
+	// }).SetupWithManager(k8sManager)
+	// Expect(err).ToNot(HaveOccurred())
```
Comment (sriram-gn) [inline controllers/suite_test.go:120]: Please remove if not used.

File: controllers/host/platformnetworks_test.go [hunk 6]
```diff
@@ -0,0 +1,646 @@
+package host
+
+import (
+	// "context"
```
Comment (sriram-gn) [inline controllers/host/platformnetworks_test.go:4]: Please remove if not used.

File: controllers/host/platformnetworks_test.go [hunk 7]
```diff
@@ -0,0 +1,646 @@
...
+
+	if host_instance.Status.StrategyRequired == cloudManager.StrategyLockRequired {
+		HostsListBodyResponse = strings.Replace(HostsListBody, `"administrative": "unlocked",`, `"administrative": "locked",`, 1)
+		HostsListBodyResponse = strings.Replace(HostsListBodyResponse, `"operational": "enabled"`, `"operational": "disabled"`, 1)
+	} else {
+		HostsListBodyResponse = HostsListBody
+	}
+}
+
+var _ = Describe("Networking utils", func() {
```
Comment (sriram-gn) [inline controllers/host/platformnetworks_test.go:241]: Networking utils -> Platform network sub-reconciler

File: controllers/host/platformnetworks_test.go [hunk 8]
```diff
@@ -0,0 +1,646 @@
...
+						Name:      pool.Name,
+						Namespace: TestNamespace,
+					}
+
+					fetched_pool := &starlingxv1.AddressPool{}
+					Eventually(func() bool {
+						err := k8sClient.Get(ctx, key_pool, fetched_pool)
+						return err == nil &&
+							fetched_pool.ObjectMeta.ResourceVersion != expected_addrpool.ObjectMeta.ResourceVersion &&
+							fetched_pool.Status.Reconciled == true &&
```
Comment (sriram-gn) [inline controllers/host/platformnetworks_test.go:509]: Remove IPv6 pxeboot address pool from API fixtures and update this to `fetched_pool.Status.Reconciled == false` to match the test case description.

File: controllers/host/platformnetworks_test.go [hunk 9]
```diff
@@ -0,0 +1,646 @@
...
+					Namespace: TestNamespace,
+				}
+				Expect(k8sClient.Create(ctx, platform_networks[nwk_name])).To(Succeed())
+
+				expected_platformnetwork := platform_networks[nwk_name].DeepCopy()
+
+				for _, pool := range address_pools[nwk_name] {
+					IntroduceAddrPoolChange(pool)
+
+					Expect(k8sClient.Create(ctx, pool)).To(Succeed())
```
Comment (sriram-gn) [inline controllers/host/platformnetworks_test.go:606]: I don't see the spec modified to remove any of 'floatingAddress', controller0Address' or 'controller1Address'.
Can you do it before creating the addresspool resource?

File: controllers/host/platformnetworks_test.go [hunk 10]
```diff
@@ -0,0 +1,646 @@
...
+	Context(" Verify reconfiguration is blocked for all networks in Day-1", func() {
+		It("Should be created successfully and Reconciled should be true & InSync should be 'false'", func() {
+			tMgr := cloudManager.GetInstance(k8sManager)
+			HostControllerAPIHandlers()
+			tMgr.SetSystemReady(TestNamespace, true)
+
+			CreateDummyHost("controller-0")
+			defer DeleteDummyHost("controller-0")
+
+			network_names := []string{"mgmt", "admin", "oam"}
```
Comment (sriram-gn) [inline controllers/host/platformnetworks_test.go:252]: Can you add "cluster-host" to it?

File: controllers/host/platformnetworks_test.go [hunk 11]
```diff
@@ -0,0 +1,646 @@
...
+		HostsListBodyResponse = strings.Replace(HostsListBodyResponse, `"operational": "enabled"`, `"operational": "disabled"`, 1)
+	} else {
+		HostsListBodyResponse = HostsListBody
+	}
+}
+
+var _ = Describe("Networking utils", func() {
+
+	Context(" Verify reconfiguration is blocked for all networks in Day-1", func() {
+		It("Should be created successfully and Reconciled should be true & InSync should be 'false'", func() {
```
Comment (sriram-gn) [inline controllers/host/platformnetworks_test.go:244]: Change "Should be created successfully and Reconciled should be true & InSync should be 'false'"

to

"Should be created successfully and Reconciled should be true & InSync should be 'false' for address pools"

Make sure to specify "address pools" in all of the test cases.

**Review by sriram-gn — CHANGES_REQUESTED** [review]

File: controllers/host/platformnetworks_test.go [hunk 12]
```diff
@@ -0,0 +1,646 @@
+package host
```
Comment (sriram-gn) [inline controllers/host/platformnetworks_test.go:1]: Add following copyright notice:

```
/* SPDX-License-Identifier: Apache-2.0 */
/* Copyright(c) 2024 Wind River Systems, Inc. */
```

**Review by sriram-gn — CHANGES_REQUESTED** [review]

File: controllers/host/host_suite_test.go (hunk 1 above)
Comment (MahalakshmiManthiram) [reply controllers/host/host_suite_test.go:2]: Resolved the comment

File: controllers/platformnetwork_controller_test.go (hunk 2 above)
Comment (MahalakshmiManthiram) [reply controllers/platformnetwork_controller_test.go:14]: Removed

File: controllers/platformnetwork_controller_test.go (hunk 3 above)
Comment (MahalakshmiManthiram) [reply controllers/platformnetwork_controller_test.go:16]: Removed

File: controllers/suite_test.go (hunk 5 above)
Comment (MahalakshmiManthiram) [reply controllers/suite_test.go:120]: Removed

File: controllers/host/platformnetworks_test.go (hunk 11 above)
Comment (MahalakshmiManthiram) [reply controllers/host/platformnetworks_test.go:244]: Changed as requested

File: controllers/host/platformnetworks_test.go (hunk 12 above)
Comment (MahalakshmiManthiram) [reply controllers/host/platformnetworks_test.go:1]: Added the copyright

File: controllers/host/platformnetworks_test.go (hunk 9 above)
Comment (MahalakshmiManthiram) [reply controllers/host/platformnetworks_test.go:606]: Changes are resolved

**Comment by sriram-gn** [comment]
Please close this pull request, I've fixed and raised another PR on your behalf which can be readily merged to master as well here:
https://github.com/Wind-River/cloud-platform-deployment-manager/pull/366

Thanks.

**Pull request closed without merging**